crates/consensus/beacon/src/engine/mod.rs

```diff
@@ -555,10 +568,29 @@ where
             } else {
                 state.head_block_hash
             };
-            self.sync.set_pipeline_sync_target(target);
+
+            // we need to first check the buffer for the head and its ancestors
+            let lowest_unknown_hash = self
+                .blockchain
+                .lowest_buffered_ancestor(target)
+                .map(|block| block.parent_hash)
+                .unwrap_or(target);
+
+            trace!(target: "consensus::engine", request=?lowest_unknown_hash, "Triggering pipeline with target instead of downloading");
+
+            self.sync.set_pipeline_sync_target(lowest_unknown_hash);
         } else {
+            // we need to first check the buffer for the head and its ancestors
+            let lowest_unknown_hash = self
+                .blockchain
+                .lowest_buffered_ancestor(state.head_block_hash)
+                .map(|block| block.parent_hash)
+                .unwrap_or(state.head_block_hash);
+
+            trace!(target: "consensus::engine", request=?lowest_unknown_hash, "Triggering full block download for the new head");
+
             // trigger a full block download for the _missing_ new head
```

Comment:
this is no longer accurate, I think, this is now, the lowest unknown hash which is not necessarily the missing head?